### ISSUE TITLE
Sending SIGHUP to consumer results in error in Consumer::restartDaemon()

### ISSUE THREAD

**Issue by joorloohuis** [mikemadisonweb/yii2-rabbitmq#42]
Hi Mikhail,

Sending a SIGHUP to a consumer results in an error:
```
2019-02-13 11:54:34 [-][-][-][error][Error] Error: Call to undefined method PhpAmqpLib\Connection\AMQPLazyConnection::renew() in /app/mikemadisonweb/yii2-rabbitmq/components/BaseRabbitMQ.php:69
Stack trace:
#0 /app/mikemadisonweb/yii2-rabbitmq/components/Consumer.php(241): mikemadisonweb\rabbitmq\components\BaseRabbitMQ->renew()
#1 [internal function]: mikemadisonweb\rabbitmq\components\Consumer->restartDaemon(1, Array)
#2 /app/php-amqplib/php-amqplib/PhpAmqpLib/Wire/IO/StreamIO.php(233): pcntl_signal_dispatch()
```
It looks like `BaseRabbitMQ::renew()` should be calling `$this->conn->reconnect()` in stead of `$this->conn->renew()`.

Joor

**Comment by mikemadisonweb** [comment]
Hi Joor,
Seems like this one is not that hard to fix, i will check and change the method.

**Comment by joorloohuis** [comment]
I made the change manually in my local vendor folder for the project I'm working on, and I see my consumers reloading perfectly.

**Comment by mikemadisonweb** [comment]
I have pushed the fixes for this one and #40 as well to dev-master. Could you verify it as well before I release? For some reason, it seems that consumer is only handling SIGHUP once, then nothing would happen for any signal (only SIGKILL could stop it afterwords).

**Comment by joorloohuis** [comment]
Sure, I'll look into this over the weekend.

**Comment by joorloohuis** [comment]
I can confirm that consumers no longer respond to signals after the first restart (SIGHUP). I haven't been able to really dig into this, but it has to do with calling Consumer::consume() during the restart. When the call to startConsuming() is done, the process no longer responds to signals. If startConsuming() is not called during a restart, the process does respond to signals, but of course we're no longer actally consuming. 

I haven't got the time right now to do a proper analysis, but it's a place to start.

**Comment by mikemadisonweb** [comment]
I finally had some time to fix it, so the new version(2.1.0) contains that fix. Reconnection is performed on each SIGHUP now. Only one restriction applied: Signals would be handled right after the consumed message, not in the middle of the message consume nor during waiting for the new message.
